### ISSUE TITLE
"path.relpath" on Windows is relative to the real current directory, not the fake directory

### ISSUE THREAD

**Issue by GoogleCodeExporter** [davidlbaird/pyfakefs#21]
```
What steps will reproduce the problem?

import os
import fake_filesystem

filesystem = fake_filesystem.FakeFilesystem()
fake_os = fake_filesystem.FakeOsModule(filesystem)

os.chdir('\\foo\\bar')

print "real curdir:\t%s" % os.getcwd()
print "fake curdir:\t%s" % fake_os.getcwd()

print "real relative to root:\t%s" % os.path.relpath('\\')
print "fake relative to root:\t%s" % fake_os.path.relpath('\\')

What version of the product are you using? On what operating system?
2.3 on Windows

Please provide any additional information below.
```

Original issue reported on code.google.com by `londinop` on 17 Sep 2013 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
Note this issue does not occur on Linux (posixpath)
```

Original comment by `londinop` on 17 Sep 2013 at 8:03

**Comment by GoogleCodeExporter** [comment]
```
Update: the root of this is that "abspath" doesn't work correctly on Windows, 
which "relpath" depends on.
```

Original comment by `londinop` on 17 Sep 2013 at 8:36

**Comment by GoogleCodeExporter** [comment]
Original comment by `dba...@google.com` on 17 Sep 2013 at 8:51
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Small update, I have it fixed, but now I am trying to get the unit tests to 
pass in Windows.
```

Original comment by `dba...@google.com` on 19 Sep 2013 at 9:07

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision e68a0a9ab329.
```

Original comment by `dba...@google.com` on 19 Sep 2013 at 11:36
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue is fixed in pyfakefs 2.4. All the tests now run on Windows, there is 
full Windows support in pyfakefs.
```

Original comment by `dba...@google.com` on 23 Sep 2013 at 9:49
